### DISCUSSION TITLE
Difference in beforeValidate hook V2 vs V3

### DISCUSSION THREAD

**Discussion by JeffreyArts** [payloadcms/payload#9614] [General]
In the migration of V2 to V3 I noticed that my hooks are no longer updating the admin UI in real-time. On the database level they work, but the changes are not reflected in the UI.

For example; I have a `beforeValidate` method that updates the blockName whenever it is empty. In order for me to actually see the updated blockName, I need to refresh the page in V3 in order to see the change. Where previously in V2 this worked instantly, updating the block name as soon as I hit save.

Is this a bug? Is this a breaking change? Should my code be changed? Could someone tell me more about this difference in experience between V2 & V3?

**Comment by andershermansen** [comment]
Probably this bug #9423 
It’s fixed but not released yet.

**Comment by JeffreyArts** [comment]
Looks like that indeed, thanks for informing me about that @andershermansen. I'll keep an eye on future releases
